onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -0,0 +1,336 @@
+  InsertAllocator(CreateAllocator(device_info));
+}
+
+static ONNX_NAMESPACE::ModelProto GetModelProtoFromFusedNode(const onnxruntime::GraphViewer& graph_viewer) {
+  ONNX_NAMESPACE::ModelProto model_proto;
+  auto graph_proto = model_proto.mutable_graph();
+
+  for (const auto& node : graph_viewer.Nodes()) {
+    node.ToProto(*(graph_proto->add_node()));
+  }
+
+  for (const auto& input : graph_viewer.GetInputs()) {
+    auto valueInfoProto = graph_proto->add_input();
+    *valueInfoProto = input->ToProto();
+  }
+
+  for (const auto& output : graph_viewer.GetOutputs()) {
+    auto valueInfoProto = graph_proto->add_output();
+    *valueInfoProto = output->ToProto();
+  }
+
+  for (const auto& initializer : graph_viewer.GetAllInitializedTensors()) {
+    graph_proto->add_initializer()->CopyFrom(*initializer.second);
+  }
+
+  auto opset = model_proto.add_opset_import();
+  opset->set_domain(kOnnxDomain);
+  opset->set_version(graph_viewer.DomainToVersionMap().at(kOnnxDomain));
+  model_proto.set_ir_version(ONNX_NAMESPACE::Version::IR_VERSION);
+
+  return model_proto;
+}
+
+std::vector<std::unique_ptr<ComputeCapability>> OpenVINOExecutionProvider::GetCapability(
+    const onnxruntime::GraphViewer& graph_viewer,
+    const std::vector<const KernelRegistry*>& /*kernel_registries*/) const {
+  std::vector<std::unique_ptr<ComputeCapability>> result;
+  bool precision_fp32 = true;
+  std::string device_id = "CPU";
+
+#ifdef OPENVINO_CONFIG_GPU_FP32
+  device_id = "GPU";
+#endif
+
+#ifdef OPENVINO_CONFIG_GPU_FP16
+  precision_fp32 = false;
+  device_id = "GPU";
+#endif
+
+#ifdef OPENVINO_CONFIG_MYRIAD
+  precision_fp32 = false;
+  device_id = "MYRIAD";
+#endif
+
+#ifdef OPENVINO_CONFIG_VAD_R
+  precision_fp32 = false;
+  device_id = "HDDL";
+#endif
+
+  auto domain_map = graph_viewer.DomainToVersionMap();
+  int opset_version = 0;
+  auto it = domain_map.find(kOnnxDomain);
+  if (it != domain_map.end())
+    opset_version = it->second;
+
+  std::vector<onnxruntime::NodeIndex> group;
+  int counter = 0;
+  auto initializers = graph_viewer.GetAllInitializedTensors();
+  auto node_indexes = graph_viewer.GetNodesInTopologicalOrder();
+  std::unique_ptr<IndexedSubGraph> sub_graph = std::make_unique<IndexedSubGraph>();
+  auto model_proto = GetModelProtoFromFusedNode(graph_viewer);
+  auto graph_proto = model_proto.mutable_graph();
+  int input_dims = 0;
+  int output_dims = 0;
+
+  if (graph_viewer.GetInputs().size() != 0)
+    input_dims = graph_proto->input(0).type().tensor_type().shape().dim_size();
+
+  if (graph_viewer.GetOutputs().size() != 0)
+    output_dims = graph_proto->output(0).type().tensor_type().shape().dim_size();
+
+  //GPU Plugin does not support single dimensional input
+  if (device_id == "GPU") {
+    if (input_dims == 1 || input_dims == 5 || output_dims == 5)
+      return result;
+  }
+
+  // bool newSubGraph(true);
+  bool isGraphSupported = true;
+  for (auto index : node_indexes) {
+    auto node = graph_viewer.GetNode(index);
```

Comment:
GetNode() can potentially return a nullptr, add a check for nullptr and skip if encountered

Reply:
please address this comment?

Reply:
Done